postgrest_py/base_request_builder.py

```diff
@@ -90,21 +86,54 @@ def pre_delete(
     return RequestMethod.DELETE, {}
 
 
-def process_response(
-    session: Union[AsyncClient, SyncClient],
-    r: Response,
-) -> Tuple[Any, Optional[int]]:
-    count = None
-    prefer_header = session.headers.get("prefer")
-    if prefer_header:
+class APIResponse(BaseModel):
+    data: Any
+    count: Optional[int] = None
+
+    @validator("data")
+    @classmethod
+    def raise_when_api_error(cls: Type[APIResponse], value: Any) -> Any:
+        if isinstance(value, dict) and value.get("message"):
+            raise ValueError("You are passing an API error to the data field.")
+        return value
+
+    @staticmethod
+    def get_count_from_content_range_header(
+        content_range_header: str,
+    ) -> Optional[int]:
+        content_range = content_range_header.split("/")
+        if len(content_range) < 2:
+            return None
+        return int(content_range[1])
+
+    @staticmethod
+    def is_count_in_prefer_header(prefer_header: str) -> bool:
         pattern = f"count=({'|'.join([cm.value for cm in CountMethod])})"
-        count_header_match = search(pattern, prefer_header)
-        content_range_header = r.headers.get("content-range")
-        if count_header_match and content_range_header:
-            content_range = content_range_header.split("/")
-            if len(content_range) >= 2:
-                count = int(content_range[1])
-    return r.json(), count
+        return bool(search(pattern, prefer_header))
+
+    @classmethod
+    def get_count_from_http_request_response(
+        cls: Type[APIResponse],
+        request_response: RequestResponse,
+    ) -> Optional[int]:
+        prefer_header: Optional[str] = request_response.request.headers.get("prefer")
+        if not prefer_header:
+            return None
+        is_count_in_prefer_header = cls.is_count_in_prefer_header(prefer_header)
+        content_range_header: Optional[str] = request_response.headers.get(
+            "content-range"
+        )
+        if is_count_in_prefer_header and content_range_header:
+            return cls.get_count_from_content_range_header(content_range_header)
+        return None
```

Comment:
Unnecessary return of None, by default all execution of functions without explicit return instruction return None.

Reply:
My reference was the following: [https://stackoverflow.com/questions/15300550/return-return-none-and-no-return-at-all](https://stackoverflow.com/questions/15300550/return-return-none-and-no-return-at-all)